hypervisor/arch/x86/guest/vioapic.c

```diff
@@ -629,17 +629,16 @@ int vioapic_mmio_access_handler(struct vcpu *vcpu, struct io_request *io_req,
 	return ret;
 }
 
-bool vioapic_get_rte(struct vm *vm, uint8_t pin, union ioapic_rte *rte)
+/**
+ * @pre vm->arch_vm.virt_ioapic != NULL
+ * @pre rte != NULL
+ */
+void vioapic_get_rte(struct vm *vm, uint8_t pin, union ioapic_rte *rte)
 {
 	struct vioapic *vioapic;
 
 	vioapic = vm_ioapic(vm);
-	if ((vioapic != NULL) && (rte != NULL)) {
```

Comment:
You removed this, so assume it won't hit. Please add pre-assumption in API description.